flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/PushProjectIntoTableSourceScanRule.java

```diff
@@ -337,11 +339,69 @@ private RowType performPushDown(
     private List<RexNode> rewriteProjections(
             RelOptRuleCall call, TableSourceTable source, NestedSchema projectedSchema) {
         final LogicalProject project = call.rel(0);
+        List<RexNode> newProjects = project.getProjects();
+
         if (supportsProjectionPushDown(source.tableSource())) {
-            return NestedProjectionUtil.rewrite(
-                    project.getProjects(), projectedSchema, call.builder().getRexBuilder());
-        } else {
-            return project.getProjects();
+            // if support project push down, then all input ref will be rewritten includes metadata
```

Comment:
Since `NestedProjectionUtil` has handled metadata column and physical column, we should use this tool instead of a new one

Reply:
After several failure cases debuging and an offline discussion with @godfreyhe and @fsk119, we decide rollback this fix to the original version due to the design assumption of `NestedProjectionUtil` which relies on the rewritten of the input row type.

Reply:
I've created a ticket to track this: https://issues.apache.org/jira/browse/FLINK-28434
cc @fsk119